netket/graph/space_group.py

```diff
@@ -196,7 +204,7 @@ def _little_group_index(self, k: Array) -> Array:
         is_in_little_group = np.all(big_star == big_star[0], axis=1)
         return np.arange(len(self.point_group_))[is_in_little_group]
 
-    def little_group(self, k: Array) -> PointGroup:
+    def little_group(self, *k) -> PointGroup:
```

Comment:
This is still fine (https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values):
```suggestion
    def little_group(self, *k: Array) -> PointGroup:
```

Reply:
Are you sure? the PEP you linked sounds like k would then be a `Tuple` of `Array`s which is definitely not what we want. Also, we support non-`Array`s like lists/tuples of numbers, so the original annotation wasn't strictly correct

Reply:
Argument lists are tuples, whether you want it or not: https://docs.python.org/3/tutorial/controlflow.html#arbitrary-argument-lists 

The type annotation has no influence on that (it's just an annotation after all).

@PhilipVinc I don't recall: Is there a type we commonly use for array-like arguments? That might be appropriate here then, if this function also accepts array-like arguments.

Reply:
(@femtobit is right: those annotations are just here to generate nicer documentation on the web and to make the code more clear. I want them because they make the code more maintanable, but they are just... comments)

IIRC `Array` is defined to be a jax/nmpy array. 
We should have a type for an Iterable (i think there is, in typing, and is called iterable)

Reply:
I'm thinking more like https://numpy.org/devdocs/reference/typing.html#numpy.typing.ArrayLike. (`Iterable` would not work, because `(j)np.asarray` should work on it and that does _not_ work for a generic iterable. It needs at least a `Sequence`.)

Reply:
If we only want this to be morally correct and readable in the docs, `Array` is probably the best, we understand that `np.tensordot` uses every meaningful array-like the right way.

Otherwise it would have to be a `Union` of scalars (so it matches the vararg calling sequence) and array-likes (@femtobit does `ArrayLike` exist in JAX?)

Reply:
JAX doesn't really use particularly specific type annotations for arrays:
https://github.com/google/jax/blob/e5e5bb3ac1019f0b57097ec9aa77f60ba826844f/jax/_src/lax/lax.py#L68
```
Array = Any
```

I'd say we stay with our NetKet `Array`, which is defined here:
https://github.com/netket/netket/blob/89f9a47b11df05f7c3a0a6bb97a8b71c098fc1b0/netket/utils/types.py#L39

This is not really true, since we can also sometimes accept `list` or similar, but `Array` seems safest and conveys the intent. Also, this is what we had here before.

Potentially cleaning this up with `ArrayLike` or similar is something to discuss for another time.

Reply:
Okay, that's what I've just done.